core/src/commonMain/kotlin/Event.kt

```diff
@@ -0,0 +1,45 @@
+package com.juul.kable
+
+public sealed class Event {
+
+    /** Triggered upon a connection being successfully established. */
+    public data class Connected(
+        val peripheral: Peripheral,
+    ) : Event()
+
+    /**
+     * Triggered either immediately after an established connection has dropped or after a failed
+     * connection attempt.
+     *
+     * @param wasConnected is `true` if event follows an established connection, or `false` if previous connection attempt failed.
+     */
+    public data class Disconnected(
+        val wasConnected: Boolean,
```

Comment:
The use-case for solely a `Disconnected` is reconnections with exponential backoff.

Having a single event that you can react to makes it trivial to implement reconnect logic. The `wasConnected` makes it easy to determine if the backoff delay should be reset.

If a connection fails, it will be propagated from the `Peripheral.connect` suspend function.

Reply:
So if a connection fails, the `Peripheral.connect` will throw, and a `Disconnected` event will also be emitted? Do I have that right?

It's cool if so, am just wanting to fully understand what happens.

Reply:
Yes, you have that right.

Reply:
When connect fails, 3 things will actually occur:

- `Peripheral.connect` will throw
- `Peripheral.state` will hold/emit a `Disconnected` state
- `Peripheral.event` will emit a `Disconnected` event

Reply:
The difference between state and events is outlined here: https://github.com/JuulLabs/able/tree/develop/keep-alive#status